src/sinks/util/sink.rs

```diff
@@ -464,27 +469,33 @@ where
         let partition = item.partition();
 
         if let Some(batch) = self.partitions.get_mut(&partition) {
-            if batch.len() >= self.settings.size {
+            if batch.was_full() {
                 trace!("Batch full; driving service to completion.");
                 self.poll_complete()?;
 
                 if let Some(batch) = self.partitions.get_mut(&partition) {
-                    if batch.len() >= self.settings.size {
+                    if !batch.is_empty() {
```

Comment:
This is a little confusing, but I get that it's in combination with the fullness check above.

Reply:
With the change in the semantics of "full", it's no longer possible to ask a buffer if it is full, only if adding something would overflow, because a buffer that tracks bytes could both be full (can't add the next event) and not full (has more bytes left in the buffer) at the same time. That path leads to insanity. However, since the buffer is fully emptied on send, the old logic of `is_full` is equivalent to `!is_empty`. I'll add an explanatory note.